common/darkside.go

```diff
@@ -248,7 +294,12 @@ func DarksideApplyStaged(height int) error {
 		}
 		block[68]++ // hack HashFinalSaplingRoot to mod the block hash
 		block = append(block, tx.bytes...)
-		state.activeBlocks[tx.height-state.startHeight] = block
+		state.activeBlocks[tx.height-state.startHeight].bytes = block
+		// Now increment this and every subsequent block's commitment tree sizes.
+		for _, b := range state.activeBlocks[tx.height-state.startHeight:] {
+			b.saplingTreeSize += uint32(tx.saplingOutputs)
+			b.orchardTreeSize += uint32(tx.orchardActions)
+		}
```

Comment:
In local testing I'm getting some off-by-twos, and while I haven't identified the source yet, I expect I have made a mistake in this change.